src/features/menu/components/Menu/Menu.tsx

```diff
@@ -47,7 +47,7 @@ export const Menu = () => {
       <nav className={`w-full h-[72px] flex justify-center items-center gap-[38px] bg-[${menuBGColor}]`}>
         {links.map((item) => {
           const upperCased = item.name[0].toUpperCase() + item.name.slice(1);
-          const isIconFilled = pathname === `/${item.name}`;
+          const isIconFilled = pathname.slice(0, item.name.length + 1).includes(item.name);
```

Comment:
Hmm, i think we might to make it another way. You have the links array, just add a new field, like:
```ts
{
  name: string
  icon: React.FC<{ fill: string }>
  link: string // Path: /path-1/:id/path-2
  linkRegex: RegExp // RegExp: /^\/path-1\/([^/]+)\/path-2$/
}

// And call: linkRegex.test(pathname)
```

Or create a new sub-component for `Menu → MenuLink` (in my opinion more accurate):
```tsx
type Props = {
  link: string
  name: string
  icon: React.FC<{ fill: string }>
}

export const MenuLink: React.FC<Props> = (props) => {
  const { link, name, icon: Icon } = props
  const { isExact: isActiveLink } = useRouteMatch(link);
  
  return (
    <BottomMobNavLink link={link} key={name}>
      <Icon fill={isActiveLink ? "#198d99" : "#757474"} />
      <span className="text-[10px] leading-[12px] font-semibold uppercase">
        {name}
      </span>
    </BottomMobNavLink>
  );
}

// Example: https://codesandbox.io/s/react-router-useroutematch-hooks-w2p5g?file=/src/App.js
// Line 55
```

Reply:
```tsx
export type MenuLinkParams = {
  name: string
  icon: React.FC<{ fill: string }>
  link: string
  linkRegex: RegExp
}

type Props = {
  menuLink: MenuLinkParams
}